app/domain/search/audit_filter.rb

```diff
@@ -0,0 +1,42 @@
+class Search
+  class AuditFilter
+    AUDIT_STATUSES_IDENTIFIERS = [
+      :audited,
+      :non_audited,
+    ].freeze
+
+    attr_accessor :status
+
+
+    def initialize(status)
+      self.status = status ? status.to_sym : nil
+
+      raise_if_unrecognised_audit_status
+    end
+
+    def apply(scope)
+      if self.status == :audited
+        scope.joins(:audit)
+      elsif self.status == :non_audited
+        scope.left_outer_joins(:audit).where("content_items.content_id not IN( select content_id from audits)")
+      else
+        scope
+      end
+    end
+
+    def raise_if_unrecognised_audit_status
+      if status && !AUDIT_STATUSES_IDENTIFIERS.include?(status)
+        raise AuditStatusError, "unrecognised audit status: #{status}"
+      end
+    end
+
+    def self.all_status
+      [
+        OpenStruct.new(id: 'audited', name: "Audited"),
+        OpenStruct.new(id: 'non_audited', name: "Non Audited"),
```

Comment:
Not a fan of `OpenStruct` either. In this case the test coverage is very good, so it is not very likely to have something failing in production. Let's pair to find a better abstraction.